### PR TITLE
Add support for running ICMS on DBT Platform. 

### PR DESCRIPTION
**Pull request by MattHolmes123**
Environment variables are loaded via CloudFoundryEnvironment or DBTPlatformEnvironment.
settings.py remains agnostic to the environment in which it is running.
When building the container image the fake environment variables in .env.circleci are used to ensure there are no validation errors.

### CODE REVIEW

File: config/settings.py [hunk 1]
```diff
@@ -200,11 +200,7 @@
 # File storage
 # for https://github.com/uktrade/django-chunk-s3-av-upload-handlers
 
-if env.vcap_services:
-    app_bucket_creds = env.vcap_services.aws_s3_bucket[0]["credentials"]
-else:
-    app_bucket_creds = {}
-
+app_bucket_creds = env.get_s3_bucket_config()
```
Comment (chris-pettinga) [inline config/settings.py:203]: I know it's not open for review yet but I wonder if we should use @computed_value for the env.py file so we can access them as properties much like the other env vars

**Review by chris-pettinga — APPROVED** [review]
Need to get more comfortable with DBT PaaS but looks good to me

**Comment by acodeninja** [comment]
Looking good, I might point some other teams to this as a good example of config management for their migrations. We will have a few other bits for you to get in, but waiting on a review of the migration docs before that is publicly available.

**Comment by MattHolmes123** [comment]
> Looking good, I might point some other teams to this as a good example of config management for their migrations. We will have a few other bits for you to get in, but waiting on a review of the migration docs before that is publicly available.

Cool sounds good @acodeninja I will leave this unmerged until it's finished then.

**Comment by MattHolmes123** [comment]
Can confirm it loads the correct environment for cloud foundry when deployed.
image

File: requirements-base.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 Authlib==1.2.1
 boto3==1.29.6
 celery[redis]==5.3.6
+dbt-copilot-python==0.1.3
```
Comment (MattHolmes123) [inline requirements-base.txt:4]: Will upgrade this when I do the celery healthcheck change

**Pull request merged**